graphql_compiler/cost_estimation/filter_selectivity_utils.py

```diff
@@ -44,6 +51,138 @@ def _are_filter_fields_uniquely_indexed(filter_fields, unique_indexes):
     return False
 
 
+def _try_obtain_parameter_value(argument_name, parameters):
+    """Return value of argument in parameters dict or raise ValueError."""
+    parameter_name = get_parameter_name(argument_name)
+    if parameter_name not in parameters:
+        raise ValueError(u'Parameter {} not found in'
+                         u'provided parameters dict {}: {}'
+                         .format(parameter_name, parameters, argument_name))
+
+    return parameters[parameter_name]
+
+
+def _convert_uuid_string_to_int(uuid_string):
+    """Return the integer representation of a UUID string."""
+    return UUID(uuid_string).int
+
+
+def _get_selectivity_of_integer_inequality_filter(
+    domain_interval, parameter_values, filter_operator
+):
+    """Return the selectivity of a given integer inequality filter filtering over a given interval.
+
+    Args:
+        domain_interval: tuple of (int, int), describing the inclusive lower and upper bound of the
+                         domain of integers being filtered.
+        parameter_values: tuple of integers, describing the parameters for the inequality filter.
+        filter_operator: str, describing the inequality filter operation being performed.
+
+    Returns:
+        Selectivity object, describing the selectivity of the integer inequality filter.
+    """
+    # Inequality filters on integers can also be represented by an interval for the subset of values
+    # that pass through the filter. By representing the filter as an interval, we avoid considering
+    # the operators separately.
+    query_interval = (domain_interval[0], domain_interval[1])
+    if filter_operator == '>':
```

Comment:
All the`[0]` and `[1]` here are making this code quite difficult to read and understand. I'd suggesting splitting the code into two smaller functions, one that handles binary operators like `>/>=/</<=`, and one for `between` (and potentially other ternary operators in the future). Then, you can do `parameter_values[0]` with an assert in the first case, and `low_value, high_value = parameter_values` for the second case.

With that structure, the reader only has to verify that the `min` and `max` calls contain the same thing on the left-hand side of the `=` as they do inside the `min/max` call.